iot_config/src/gateway_service.rs

```diff
@@ -59,6 +69,25 @@ impl GatewayService {
             .map_err(|_| Status::permission_denied("invalid admin signature"))?;
         Ok(())
     }
+
+    async fn resolve_gateway_info(&self, pubkey: &PublicKeyBinary) -> Result<GatewayInfo, Status> {
```

Comment:
i'm curious about thoughts on having this instead return `Result<(GatewayInfo, bool), Status>` where the bool is whether or not we had a cache hit. since this function is hiding the result of that call it means i lose the ability to measure in the request duration histogram metric (different PR) whether or not a request was processed through the cache via a label